docs/user/troubleshooting.md

```diff
@@ -0,0 +1,91 @@
+machine-api-operator-7894d8f85-lq2ts             1/1       Running   0          4h
+```
+
+The logs for the machine-controller container within the `clusterapi-manager-controllers` pod need to be checked to determine why the workers haven't been created. That can be done with the following (the exact name of the pod will need to be substituted):
+
+```sh
+oc --config=${INSTALL_DIR}/auth/kubeconfig --namespace=openshift-cluster-api logs clusterapi-manager-controllers-774dc4557-nx5xq --container=machine-controller
+```
+
+### API is Unavailable
+
+When the API is unavailable, the master nodes will need to checked to ensure that they are running the correct components. This requires SSH access so it is necessary to include an administrator's SSH key during the installation.
+
+If SSH access to the master nodes isn't available, that will need to be [investigated next](#unable-to-ssh-into-master-node).
+
+The first thing to check is to make sure that etcd is running on each of the masters. The etcd logs can be viewed by running the following on each master node:
+
+```sh
+sudo crictl logs $(sudo crictl ps --pod=$(sudo crictl pods --name=etcd-member --quiet) --quiet)
+```
+
+If the previous command fails, ensure that the etcd pods have been created by the Kubelet:
+
+```sh
+sudo crictl pods --name=etcd-member
+```
+
+If no pods are shown, etcd will need to be [investigated](#etcd-is-not-running).
+
+### Unable to SSH into Master Nodes
+
+In order to SSH into the master nodes, it is necessary to include an administrator's SSH key during the installation. If SSH authentication is failing, ensure that the proper SSH key is being used.
+
+If SSH isn't able to connect to the nodes, they may be waiting on the bootstrap node before they can boot. The initial set of master nodes fetch their boot configuration (the Ignition Config) from the bootstrap node and will not complete until they successfully do so. Check the console output of the nodes to determine if they have successfully booted or if they are waiting for Ignition to fetch the remote config.
+
+Master nodes waiting for Ignition is indicative of problems on the bootstrap node. SSH into the bootstrap node to [investigate further](#unable-to-ssh-into-bootstrap-node).
+
+### Unable to SSH into Bootstrap Node
+
+If the bootstrap node isn't available, first double check that it hasn't been automatically removed by the installer. If it's not being created in the first place, the installer will need to be [troubleshot](#installer-fails-to-create-resources).
+
+After using SSH to access the bootstrap node, the most important thing to look at is `bootkube.service`. The logs can be viewed with the following command:
+
+```sh
+journalctl --unit=bootkube.service
+```
+
+### etcd Is Not Running
+
+etcd is started and managed by the Kubelet as a static pod. This requires a newer Kubelet which started shipping with version 47.29 of Red Hat CoreOS. The OS version can be checked using the following command:
```

Comment:
I think it would be more correct to specify installer version, as it is the one which instructs kubelet to run it as a static pod, instead of previous podman run

Reply:
> I think it would be more correct to specify installer version, as it is the one which instructs kubelet to run it as a static pod, instead of previous podman run

The installer [sets the `kubelet.service`][1] (at least on the bootstrap node), but the `kubelet` is being provided by the OS `/usr/bin/hyperkube`, which is independent of the installer.

> Off topic, but what version of the kubelet is this?

Back around when we made the change to static-pod etcd, @abhinavdahiya asked about openshift/origin#21274 landing in RHCOS.  Maybe that helps track that down?

[1]: https://github.com/openshift/installer/blob/f4ded2511f5016be0664d740f68c379e476944ec/data/data/bootstrap/systemd/units/kubelet.service.template